### DISCUSSION TITLE
Delivering static SSG sites and fallback to SPA without 404 status-code

### DISCUSSION THREAD

**Discussion by cewald** [nuxt/nuxt#24583] [Questions]
Is it possible to prerender specific sites using e.g. a module using `addPrerenderRoutes` to fetch all available routes BUT also deliver the fallback SPA version without serving it with a 404 status-code?

What I mean is, I have a platform where not all routes need to be prerendered (as it would be a lot and most are not important) and if I prerender the important ones its fine. But I want the rest to be available using client-side-rendering (SPA mode). It actually already works and delivers the fallback correctly using the SPA but with 404 status-code. If I could switch that into a 200 everything would be fine. Help is appreciated, thanks 🙂

**Comment by TheAlexLichter** [comment]
You can also switch to a 200, yes. But then, also 404s from Vue/Nuxt will be 200s.
But changing that must happen on your platform, e.g. configure nginx/apache or set up redirects to the ´200.html`

**Reply by cewald** [reply]
I found that solution already, thank you! So, this would be "best practice", despite the fact I'd opt-out 404 pages in a SPA manner?

And is there a way to tell the Nitro preview command (`node ./server/index.mjs`) to not fallback on a 404 page, to have the same experience in local-/dev-preview?

**Reply by TheAlexLichter** [reply]
Both have their issues but I rather have "soft-404s" than 404s everywhere.

> And is there a way to tell the Nitro preview command (node ./server/index.mjs) to not fallback on a 404 page, to have the same experience in local-/dev-preview?

You could change the response code via nitro in dev (e.g. through a middleware), but AFAIK there is no default to do so
